codalab/worker/worker_run_state.py

```diff
@@ -475,6 +489,8 @@ def _transition_from_UPLOADING_RESULTS(self, run_state):
         If uploading and finished:
             Move to FINALIZING state
         """
+        if run_state.is_restaged:
+            return run_state._replace(stage=RunStage.STAGED)
```

Comment:
Do you need these two lines in so many places?  I would have thought you'd follow something analogous to `is_killed`.

Reply:
Yes. That's what I originally suggested. We had a discussion with @nelson-liu two weeks ago to decide in what state a bundle can be moved from worker to `staged` state. @nelson-liu suggested allowing bundles in any states except for the final state to be able to move back to the `staged` state. And adding these two lines checking is to fulfill this requirement. But please let me know if this is not the case anymore.